DuckDuckGo/DuckPlayer/YouTubePlayerNavigationHandler.swift

```diff
@@ -105,6 +121,43 @@ extension YoutubePlayerNavigationHandler: DuckNavigationHandling {
                           webView: WKWebView,
                           completion: @escaping (WKNavigationActionPolicy) -> Void) {
         
+        
+        // If trying to load the same video while DP is visible
+        // Just open it in Youtube
+        if let url = navigationAction.request.url,
+           let (videoID, _) = url.youtubeVideoParams,
+           videoID == currentYoutubeVideoID {
+            isDuckPlayerTemporarilyDisabled = true
+            os_log("DP: Trying to load the same video while in DuckPlayer, use Youtube:", log: .duckPlayerLog, type: .debug)
+            webView.load(URLRequest(url: URL.youtube(videoID)))
+            completion(.allow)
+            return
+        }
+        
+        
+        os_log("DP: Handling DuckPlayer Player Navigation for %s", log: .duckPlayerLog, type: .debug, navigationAction.request.url?.absoluteString ?? "")
+       
+        // Handle Open in Youtube Links
+        // duck://player/watchInYoutube?v=12345
+        if let url = navigationAction.request.url,
```

Comment:
The "Watch in Youtube" button fires up this type of link.  This is the way Windows and Android take care of that.